pkg/usagestats/stats.go

```diff
@@ -154,27 +154,67 @@ func memstats() interface{} {
 }
 
 // NewFloat returns a new Float stats object.
+// If a Float stats object with the same name already exists it is returned.
 func NewFloat(name string) *expvar.Float {
+	existing := expvar.Get(statsPrefix + name)
+	if existing != nil {
+		if f, ok := existing.(*expvar.Float); ok {
+			return f
+		}
+		panic(fmt.Sprintf("%v is set to a non-float value", name))
+	}
 	return expvar.NewFloat(statsPrefix + name)
 }
 
 // NewInt returns a new Int stats object.
+// If an Int stats object object with the same name already exists it is returned.
 func NewInt(name string) *expvar.Int {
+	existing := expvar.Get(statsPrefix + name)
+	if existing != nil {
+		if i, ok := existing.(*expvar.Int); ok {
+			return i
+		}
+		panic(fmt.Sprintf("%v is set to a non-int value", name))
+	}
 	return expvar.NewInt(statsPrefix + name)
 }
 
 // NewString returns a new String stats object.
+// If a String stats object with the same name already exists it is returned.
 func NewString(name string) *expvar.String {
+	existing := expvar.Get(statsPrefix + name)
+	if existing != nil {
+		if s, ok := existing.(*expvar.String); ok {
+			return s
+		}
+		panic(fmt.Sprintf("%v is set to a non-string value", name))
+	}
 	return expvar.NewString(statsPrefix + name)
 }
 
-// Target sets the target name.
+// Target sets the target name. This can be set multiple times.
 func Target(target string) {
+	existing := expvar.Get(statsPrefix + targetKey)
```

Comment:
you don't need it on Target and Edition anymore since they rely on NewString

Reply:
D-oh. That code has been removed.